examples/fluids/qfunctions/newtonian.h

```diff
@@ -134,12 +134,7 @@ CEED_QFUNCTION(RHSFunction_Newtonian)(void *ctx, CeedInt Q, const CeedScalar *co
     State            s      = StateFromU(context, U, x_i);
 
     State grad_s[3];
-    for (CeedInt k = 0; k < 3; k++) {
-      CeedScalar dx_i[3] = {0}, dU[5];
-      for (CeedInt j = 0; j < 5; j++) dU[j] = Grad_q[0][j][i] * dXdx[0][k] + Grad_q[1][j][i] * dXdx[1][k] + Grad_q[2][j][i] * dXdx[2][k];
-      dx_i[k]   = 1.;
-      grad_s[k] = StateFromU_fwd(context, s, dU, x_i, dx_i);
-    }
+    StatePhysicalGradientFromReference(Q, i, context, s, x_i, STATEVAR_CONSERVATIVE, Grad_q, dXdx, false, grad_s);
```

Comment:
Is it possible/desirable to use `CEED_Q_VLA` here instead of just `Q`? With the current implementation, the compiler knows the exact offset via `CEED_Q_VLA` being set. I'd imagine that for GPUs it could possibly compile better if it knew the exact offset that would be required to get to each component of `Grad_q`.

Reply:
That value of `Q` is a compile time constant set during the code generation:
```
  code << "  const CeedInt Q_1d = " << Q_1d << ";\n";
```

Reply:
So it's basically equivalent to `CEED_Q_VLA`?

Reply:
After all the compilation optimization passes, I think its functionally the same. `CEED_Q_VLA`  drops in an integer literal for the GPU JiT to get around the lack of support for VLAs in C++ blocking our recasting for readable indexing. But the JiT compiler will know that `Q` isn't changing because its based on a const value and should be able to optimize to the same place.

Reply:
I suppose we could use a macro that uses `CEED_Q_VLA` so it doesn't have to be written and the implementation could use `CEED_Q_VLA` to avoid the more clumsy flattened indexing. The issue is that C++ doesn't have VLA (VLA pointers) so we can't dimension based on the "variable" `Q` even when the compiler will specialize it based on known compile-time values. Similarly, this is not valid C++
```cpp
const int n = 3;
int x[n];
```

Reply:
> I suppose we could use a macro that uses `CEED_Q_VLA` so it doesn't have to be written and the implementation could use `CEED_Q_VLA` to avoid the more clumsy flattened indexing.

I'm not sure I follow. Do you mean converting `StatePhysicalGradientFromReference*` to be a macro rather than a helper?

Reply:
If you did that, then it could use `CEED_Q_VLA`. I don't have a strong opinion either way, and expect that what you did here has no impact on performance.

Reply:
I'll double check the performance real quick. I highly doubt it'll change either.

Reply:
Everything looks good to me. If anything, this branch is slightly faster than `main` right now for some reason (after rebaseing off of main of course).